### PR TITLE
PoS Change ForkHeight Checks to Use DeSoParams and Simplify BlockView TxnConnects condition

### PR DESCRIPTION
**Pull request by AeonSw4n**

### CODE REVIEW

**Comment by AeonSw4n** [comment]
Current dependencies on/for this PR:
* `feature/pos-syncing-and-steady-state`
  * **PR #1009** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1009?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/1009?utm_source=stack-comment).

File: lib/block_view.go [hunk 1]
```diff
@@ -4058,10 +4058,26 @@ func (bav *UtxoView) ConnectBlock(
...
+		// In PoS, these failing transactions are included in the block and their fees are burned.
+
+		// To determine if we're dealing with a connecting or failing transaction, we first check if we're on a PoS block
+		// height. Otherwise, the transaction is expected to connect.
+		hasPoWBlockHeight := bav.Params.IsPoWBlockHeight(blockHeight)
+		// Also, the first transaction in the block, the block reward transaction, should always be a connecting transaction.
+		isBlockRewardTxn := false
+		if txIndex == 0 {
+			isBlockRewardTxn = txn.TxnMeta.GetTxnType() == TxnTypeBlockReward
+		}
```
Comment (tholonious) [inline lib/block_view.go:4071]: Fits nicely in one line and still easy to read
```suggestion
		isBlockRewardTxn := (txIndex == 0) && (txn.TxnMeta.GetTxnType() == TxnTypeBlockReward)
```

File: lib/block_view.go [hunk 2]
```diff
@@ -4058,10 +4058,26 @@ func (bav *UtxoView) ConnectBlock(
...
+		// Finally, if the transaction is not the first in the block, we check the TxnConnectStatusByIndex to see if
+		// it's marked by the block producer as a connecting transaction. PoS blocks should reflect this in TxnConnectStatusByIndex.
+		hasConnectingPoSTxnStatus := false
+		if txIndex > 0 && desoBlock.TxnConnectStatusByIndex != nil {
```
Comment (tholonious) [inline lib/block_view.go:4075]: May as well gate this by block height too to be extra clear that this is a PoS only feature
```suggestion
		if !hasPoWBlockHeight && txIndex > 0 && desoBlock.TxnConnectStatusByIndex != nil {
```

**Review by tholonious — COMMENTED** [review]
Looks good. Just some minor comments

**Review by tholonious — APPROVED** [review]

**Pull request merged**